piff/stats.py

```diff
@@ -0,0 +1,264 @@
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+
+"""
+.. module:: stats
+
+"""
+
+from __future__ import print_function
+import numpy as np
+
+from .psf import PSF
+
+def process_stats(config, logger):
+    """Parse the stats field of the config dict.
+
+    :param config:      The configuration dict.
+    :param logger:      A logger object for logging debug info.
+
+    :returns: an stats instance
+    """
+    import piff
+
+    if 'stats' not in config:
+        raise ValueError("config dict has no stats field")
+    config_stats = config['stats']
+
+    if 'type' not in config_stats:
+        raise ValueError("config['stats'] has no type field")
+
+    # Get the class to use for the stats
+    # Not sure if this is what we'll always want, but it would be simple if we can make it work.
+    stats_class = eval('piff.' + config_stats.pop('type'))
+
+    # Read any other kwargs in the stats field
+    kwargs = stats_class.parseKwargs(config_stats)
+
+    # Build stats object
+    stats = stats_class(**kwargs)
+
+    return stats
+
+class Statistics(PSF):
+    """The base class for getting the statistics of a set of stars.
+
+    It seems reasonable to use the PSF class as our base and build on top of that.
+    """
+
+    def stats(self, stars, logger=None, **kwargs):
+        """Perform your operation on the stars.
+
+        :param stars:       A list of StarData instances.
+        :param logger:      A logger object for logging debug info. [default: None]
+        :params kwargs:     Potential other parameters we might need to input. Images, coordinates, et cetera.
+
+        :returns:           Some kind of data vector.
+        """
+        raise NotImplemented("Derived classes must define the stats function")
+
+    def plot(self, fig=None, ax=None, logger=None, **kwargs):
+        """Make your plots.
+
+        :param fig:         A Matplotlib Figure object. [default: None]
+        :param ax:          A Matplotlib Axis object. [default: None]
+        :param logger:      A logger object for logging debug info. [default: None]
+        :params kwargs:     Potential other parameters we might need to input. Images, coordinates, et cetera.
+
+        :returns:           Some kind of data vector.
+        """
+        raise NotImplemented("Derived classes must define the plot function")
+
+class RhoStatistics(Statistics):
+    """Returns rho statistics using TreeCorr
+    """
+
+    def stats(self, stars, min_sep=1, max_sep=150, bin_size=0.1,
+              sky_coordinates=False, sep_units='rad',
+              ra_units='deg', dec_units='deg', logger=None):
```

Comment:
I think this would work better as an `__init__` method rather than a function.  Logically, I think it would make more sense to be building a `RhoStatistics` object from a `psf` object, a list of `stars` and some optional parameters.  Then rather than return things, this would set attributes: `self.logr, self.rho1`, etc. that someone could access.  Returning a bunch of numpy arrays doesn't make much sense to me.  As evidence, you don't use any of these return values in the unit test you wrote for this.